base_local_planner/src/speed_limiters/static_object_speed_limiter.cpp

```diff
@@ -0,0 +1,423 @@
+    {
+      SpeedLimiterResult result =
+          calculateAllowedLinearSpeed(distance_from_static_left, distance_from_static_right, max_allowed_linear_vel);
+      if (result.limiting)
+      {
+        if (result.speed < currLinearVel)
+        {
+          cachedMaxLinearVelocity_ = result.speed;
+          max_allowed_linear_vel = result.speed;
+        }
+      }
+    }
+
+    if (currAngularVel >= params_.min_angular_velocity_test_speed &&
+        currAngularVel <= params_.max_angular_velocity_test_speed)
+    {
+      SpeedLimiterResult result =
+          calculateAllowedAngularSpeed(distance_from_static_left, distance_from_static_right, max_allowed_angular_vel);
+      if (result.limiting)
+      {
+        if (result.speed < currAngularVel)
+        {
+          cachedMaxAngularVelocity_ = result.speed;
+          max_allowed_angular_vel = result.speed;
+        }
+      }
+    }
+
+    ROS_DEBUG_THROTTLE(0.2, "Setting max speed to %f, %f", max_allowed_linear_vel, max_allowed_angular_vel);
+    base_local_planner::StaticObjects staticObject_msg;
+    staticObject_msg.distance_from_static_left = distance_from_static_left;
+    staticObject_msg.distance_from_static_right = distance_from_static_right;
+    staticObject_pub.publish(staticObject_msg);
+
+    return true;
+  }
+
+  StaticObjectSpeedLimiter::SpeedLimiterResult StaticObjectSpeedLimiter::calculateAllowedLinearSpeed(
+      const double dist_left, const double dist_right, const double speed) const
+  {
+    SpeedLimiterData data;
+    data.dist_left_ = dist_left;
+    data.dist_right_ = dist_right;
+    data.speed_ = speed;
+    data.min_velocity_ = params_.min_linear_velocity;
+    data.min_test_distance_ = params_.min_linear_velocity_distance;
+    data.max_test_distance_ = params_.max_linear_velocity_distance;
+    data.min_test_velocity_ = params_.min_linear_velocity_test_speed;
+    data.max_test_velocity_ = params_.max_linear_velocity_test_speed;
+    data.min_reduction_ = params_.min_linear_velocity_reduction;
+    data.max_reduction_ = params_.max_linear_velocity_reduction;
+
+    return calculateAllowedSpeed(data);
+  }
+
+  StaticObjectSpeedLimiter::SpeedLimiterResult StaticObjectSpeedLimiter::calculateAllowedAngularSpeed(
+      const double dist_left, const double dist_right, const double speed) const
+  {
+    SpeedLimiterData data;
+    data.dist_left_ = dist_left;
+    data.dist_right_ = dist_right;
+    data.speed_ = speed;
+    data.min_velocity_ = params_.min_angular_velocity;
+    data.min_test_distance_ = params_.min_angular_velocity_distance;
+    data.max_test_distance_ = params_.max_angular_velocity_distance;
+    data.min_test_velocity_ = params_.min_angular_velocity_test_speed;
+    data.max_test_velocity_ = params_.max_angular_velocity_test_speed;
+    data.min_reduction_ = params_.min_angular_velocity_reduction;
+    data.max_reduction_ = params_.max_angular_velocity_reduction;
+
+    return calculateAllowedSpeed(data);
+  }
```

Comment:
what if these methods were called before a call to the reconfigure method was called?